packages/babel-parser/src/parser/expression.js

```diff
@@ -1046,11 +1060,27 @@ export default class ExpressionParser extends LValParser {
         this.state.inFSharpPipelineDirectBody = oldInFSharpPipelineDirectBody;
         return this.finishNode(node, "ArrayExpression");
       }
+      case tt.braceBarL:
+      case tt.braceHashL: {
```

Comment:
should assert `recordAndTuple` presents otherwise it is a syntax error.
```js
this.expectPlugin("recordAndTuple");
```
Can you add a test case when `recordAndTuple` is not enabled?